pennylane/drawer/tape_text.py

```diff
@@ -488,8 +492,15 @@ def tape_text(
 
         # Create initial strings for the current layer using wire and cwire fillers
         layer_str = [w_filler] * n_wires + [" "] * n_bits
-        for b in bit_map.values():
-            cur_b_filler = b_filler if min(cwire_layers[b]) < i < max(cwire_layers[b]) else " "
+        for b in range(n_bits):
+
+            cur_b_filler = " "
+            for layer_stretch in cwire_layers[b]:
+                if i <= layer_stretch[0]:
+                    break
+                if i < layer_stretch[-1]:
+                    cur_b_filler = b_filler
+                    break
```

Comment:
Instead of checking whether the current layer (`i`) is between start and end of a single list of layer indices, we need to check whether `i` is between _any_ of the start and end values of multiple lists of layer indices, corresponding to the multiple usages of the same classical wire.